src/Components/ComponentLockModel.cs

```diff
@@ -98,5 +104,16 @@ public string LockReason
         /// Gets or sets a value indicating a system lock
         /// </summary>
         public bool SystemLock => this.Type != ComponentLockType.Standard;
+
+        /// <summary>
+        /// Gets or sets the lock type by name.
+        /// </summary>
+        [JsonIgnore]
+        public string LockTypeName
+        {
+            get => this.Type.ToString();
+
+            set => this.Type = value.ToEnum<ComponentLockType>();
+        }
```

Comment:
I dont think we need this. Type serializes fine to string and if not, the enum should have the Json string serializer attribute added. This seems redundant. If the UI uses this property, let's remove it, and instead use State.